### DISCUSSION TITLE
Control_transfer returns a result with  same value as wLength

### DISCUSSION THREAD

**Discussion by symdeb** [libusb/libusb#1463] [Q&A] [answered]
After not being able to get a control transfer for an interface descriptor to work, tried a  control transfer for  the device descriptor. _transfer function actually work  ? When executing below, the error returned is always same value as wLength  

Ubuntu 23.10, 
```
g++ -ggdb usb.cpp -lusb-1.0

with 0x05
get_device_descriptor
result = 5 LIBUSB_TRANSFER_NO_DEVICE

with 0x12
get_device_descriptor
result = 18 **UNKNOWN**

with 0 
get_device_descriptor
result = 0 LIBUSB_SUCCESS / LIBUSB_TRANSFER_COMPLETED

  uint8_t bmRequestType = 0x80;   // DEVICE IS RECIPIENT
  uint8_t bRequest =0x06;         // GET_DESCRIPTOR (0x06)
  uint16_t wValue = 0x0100;       // DEVICE DESCRIPTOR
  uint16_t wIndex = 0;            // ENDPOINT 0
  unsigned char data[128];
  uint16_t wLength = 0x5;
  unsigned int timeout = 0xFFFF;
  int result;
     
   printf("get_device_descriptor\n");
   result = libusb_control_transfer(dev_handle,  bmRequestType, bRequest, wValue, wIndex, &data[0], wLength, timeout);
   printf ("result = %d %s\n", result, libusb_error_name(result));
   return result;  
```

**Comment by seanm** [comment]
What version of libusb? Could you try against git master?

**Comment by mcuee** [comment]
@symdeb 

Please check out this page when reporting issues.
https://github.com/libusb/libusb/wiki/Troubleshooting

You may want to post more info about your device, for example, using `lsusb -vvv` since you are using Linux. You can also check out libusb example `testlibusb -v` and `xusb` to learn from them.

**Comment by symdeb** [comment, marked as answer]
Thanks, found xusb and found the mistake. Assumed the return value was an error, but it's the length of the descriptor, no wonder :-)
